ang/civicase-base/services/util.service.js

```diff
@@ -0,0 +1,15 @@
+(function (angular, _, crmTs) {
+  var module = angular.module('civicase-base');
+
+  module.service('CivicaseUtil', function () {
+    this.capitalizeFirstLetterAndRemoveUnderScore = capitalizeFirstLetterAndRemoveUnderScore;
+
+    /**
+     * @param {string} string string to capitalize
+     * @returns {string} capitalized string
+     */
+    function capitalizeFirstLetterAndRemoveUnderScore (string) {
+      return (string.charAt(0).toUpperCase() + string.slice(1)).replace('_', '');
+    }
+  });
+})(angular, CRM._, window.ts);
```

Comment:
we should not have a generic service like this. This means that the file size will increase and will include a bunch of functions that are not related to each other. This will only make it harder to maintain the file.

It would be better to either leave the function where it was, or move the function to a service included in its own file.

If we still want to group it in a utility service we should have a structure similar to this:

```js
+ services
  + utils
    - utils.service.js
    - pascal-case.service.js
```

```js
// utils.service.js
var module = angular.module('civicase-base');

module.service('civicaseUtils', function (pascalCase) {
  return {
    pascalCase: pascalCase
  };
});
```

```js
// pascal-case.service.js
var module = angular.module('civicase-base');

module.factory('pascalCase', function () {
  return function pascalCase (string) {
    return _.chain(string).camelCase().capitalize().value(); // will also replace `_` and `-`.
  };
});
```

Also, the name for what you are trying to do is called "pascal case" (which is a bit shorter than the name provided) and can be accomplished using a couple of lodash functions.